core/network/impl/stream_engine.hpp

```diff
@@ -203,19 +202,32 @@ namespace kagome::network {
     }
 
     template <typename T>
-    void send(std::shared_ptr<Stream> stream, const T &msg) {
+    void send(const PeerId &peer_id,
+              const std::shared_ptr<ProtocolBase> &protocol,
+              std::shared_ptr<Stream> stream,
+              const T &msg) {
       BOOST_ASSERT(stream != nullptr);
 
       auto read_writer =
           std::make_shared<ScaleMessageReadWriter>(std::move(stream));
-      read_writer->write(msg, [wp = weak_from_this()](auto &&res) {
-        if (not res) {
-          if (auto self = wp.lock()) {
-            self->logger_->error("Could not send message, reason: {}",
-                                 res.error().message());
-          }
-        }
-      });
+      read_writer->write(
+          msg, [wp = weak_from_this(), peer_id, protocol](auto &&res) {
+            if (auto self = wp.lock()) {
+              if (res.has_value()) {
+                SL_TRACE(self->logger_,
```

Comment:
passing argument to use in logging looks weird

Reply:
I agree with you in general, but we does not have way to known that in lambda, and it is really needed for logging.
They are lightweight, and copy just into lambda. References are used in other cases.